opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -239,12 +239,138 @@ impl opentelemetry::logs::Logger for Logger {
+            );
+        };
+        let assert_telemetry_resource = |provider: &super::LoggerProvider| {
+            assert_eq!(
+                provider
+                    .config()
+                    .resource
+                    .get(TELEMETRY_SDK_LANGUAGE.into()),
+                Some(Value::from("rust"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_NAME.into()),
+                Some(Value::from("opentelemetry"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_VERSION.into()),
+                Some(Value::from(env!("CARGO_PKG_VERSION")))
+            );
+        };
+
+        // If users didn't provide a resource and there isn't a env var set. Use default one.
+        let default_config_provider = super::LoggerProvider::builder().build();
+        assert_resource(
+            &default_config_provider,
+            SERVICE_NAME,
+            Some("unknown_service"),
+        );
+        assert_telemetry_resource(&default_config_provider);
+
+        // If user provided a resource, use that.
+        let custom_config_provider = super::LoggerProvider::builder()
+            .with_config(Config {
+                resource: Cow::Owned(Resource::new(vec![KeyValue::new(
+                    SERVICE_NAME,
+                    "test_service",
+                )])),
+            })
+            .build();
+        assert_resource(&custom_config_provider, SERVICE_NAME, Some("test_service"));
+        assert_eq!(custom_config_provider.config().resource.len(), 1);
+
+        // If `OTEL_RESOURCE_ATTRIBUTES` is set, read them automatically
+        temp_env::with_var(
+            "OTEL_RESOURCE_ATTRIBUTES",
+            Some("key1=value1, k2, k3=value2"),
+            || {
+                let env_resource_provider = super::LoggerProvider::builder().build();
+                assert_resource(
+                    &env_resource_provider,
+                    SERVICE_NAME,
+                    Some("unknown_service"),
+                );
+                assert_resource(&env_resource_provider, "key1", Some("value1"));
+                assert_resource(&env_resource_provider, "k3", Some("value2"));
+                assert_telemetry_resource(&env_resource_provider);
+                assert_eq!(env_resource_provider.config().resource.len(), 6);
+            },
+        );
+
+        // When `OTEL_RESOURCE_ATTRIBUTES` is set and also user provided config
+        temp_env::with_var(
+            "OTEL_RESOURCE_ATTRIBUTES",
+            Some("my-custom-key=env-val,k2=value2"),
+            || {
+                let user_provided_resource_config_provider = super::LoggerProvider::builder()
+                    .with_config(Config {
+                        resource: Cow::Owned(Resource::default().merge(&mut Resource::new(vec![
+                            KeyValue::new("my-custom-key", "my-custom-value"),
```

Comment:
should we also add here a resource attribute not present in the OTEL_RESOURCE_ATTRIBUTES env var, to validate it is merged properly?

Reply:
Good point! Added https://github.com/open-telemetry/opentelemetry-rust/pull/1646/commits/5a450896f7bb971a0bc6a29f62cae9c56d633fe8